src/main/java/org/mule/extension/mulechain/internal/CommonUtils.java

```diff
@@ -0,0 +1,48 @@
+package org.mule.extension.mulechain.internal;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+
+public class CommonUtils {
+	
+    private static final Logger logger = LoggerFactory.getLogger(CommonUtils.class);
+
+
+    // Make it public so other classes (like helper or connection provider) can use it
+    public static ChronoUnit convertToChronoUnit(TimeUnitEnum unit) {
+        switch (unit) {
+            case SECONDS:
+                return ChronoUnit.SECONDS;
+            case MINUTES:
+                return ChronoUnit.MINUTES;
+            default:
+                return ChronoUnit.MILLIS;
+        }
+    }
+
+    // Optional utility to directly create a Duration
+    public static Duration toDuration(int timeout, TimeUnitEnum unit) {
+        return Duration.of(timeout, convertToChronoUnit(unit));
+    }
+    
+
+    public static SdkHttpClient buildHttpClientWithTimeout(Integer timeout, TimeUnitEnum unit) {
+    	
+    	logger.debug("timeout: {}", timeout);
```

Comment:
Since some updates are required, does it make sense to instead log the **_effective timeout and timeunit_** for the HTTP client, at debug level?